medusa/show/recommendations/recommended.py

```diff
@@ -130,9 +133,19 @@ def __init__(self, rec_show_prov, series_id, title, mapped_indexer, mapped_serie
         self.is_anime = False
 
         # Check if the show is currently already in the db
-        self.show_in_list = bool([show.indexerid for show in app.showList
-                                 if show.series_id == self.mapped_series_id and
-                                 show.indexer == self.mapped_indexer])
+        indexers = {mapped_indexer: mapped_series_id}
+        indexers.update({VALID_INDEXERS[indexer_name]: indexers_series_id
+                         for indexer_name, indexers_series_id
+                         in self.ids.items() if indexer_name in VALID_INDEXERS})
+
+        self.show_in_list = False
+        for show in app.showList:
+            if show.indexer in indexers and show.series_id == indexers[show.indexer]:
+                self.show_in_list = True
+                self.mapped_indexer = show.indexer
```

Comment:
By default shows are added with tvdb. But if they're already in the db, we want to reference the indexer and series_id from the library.